robot/web/web.go

```diff
@@ -274,6 +292,64 @@ func RunWebWithConfig(ctx context.Context, r robot.LocalRobot, cfg *config.Confi
 	return RunWeb(ctx, r, o, logger)
 }
 
+// Address returns the address the service is listening on.
+func (svc *webService) Address() string {
+	svc.mu.Lock()
+	defer svc.mu.Unlock()
+	return svc.addr
+}
+
+// LiteAddress returns the (unix socket) address the lite service is listening on.
+func (svc *webService) LiteAddress() string {
+	svc.mu.Lock()
+	defer svc.mu.Unlock()
+	return svc.liteAddr
+}
+
+func (svc *webService) StartLite(ctx context.Context) error {
+	svc.mu.Lock()
+	defer svc.mu.Unlock()
+
+	dir, err := os.MkdirTemp("", "viam-module-*")
+	if err != nil {
+		return errors.WithMessage(err, "module startup failed")
+	}
+	svc.liteAddr = dir + "/parent.sock"
+	oldMask := syscall.Umask(0o077)
```

Comment:
security comment

Reply:
Saw your other comment about security on modmanager, and came here to double-check things. I need to move the umask up a few lines, before the mkdirtemp to ensure someone can't out-race the loading to create a fake module socket. Can add a check for ownership as well.